drivers/kscan/Kconfig.ht16k33

```diff
@@ -3,7 +3,8 @@
 
 config KSCAN_HT16K33
 	bool "HT16K33 keyscan driver"
-	depends on HT16K33_KEYSCAN
+	default y
+	depends on DT_HAS_HOLTEK_HT16K33_KEYSCAN_ENABLED
```

Comment:
This one still depends on `CONFIG_ HT16K33_KEYSCAN` which is exposed by the HT16K33 LED driver.

Reply:
Yeah, I was trying to understand that.  Wasn't sure what made the most sense here.

Reply:
We could make it `select` the LED driver + key scan option similar as to how other "busses" are selected.

Reply:
@henrikbrixandersen do you mind if we merge this as is, and do a follow up PR after the LED PR (https://github.com/zephyrproject-rtos/zephyr/pull/48827) goes it for dealing with both sides at one time?

Reply:
Fine by me.